operations.go

```diff
@@ -585,3 +587,22 @@ func (c CmdTimeoutUpdate) Resolve() *CommandDefinition {
 	}
 }
 func timeoutUpdateFactory() Command { return CmdTimeoutUpdate{} }
+
+type CmdGitHubGenerateToken struct {
+	Owner         string `json:"owner,omitempty" yaml:"owner,omitempty"`
+	Repo          string `json:"repo,omitempty" yaml:"repo,omitempty"`
+	ExpansionName string `json:"expansion_name,omitempty" yaml:"expansion_name,omitempty"`
+	// Permissions is a map of permissions to grant to the token. If this is nil,
+	// the token will have all the permissions of the GitHub app.
+	Permissions *github.InstallationPermissions `json:"permissions,omitempty" yaml:"permissions,omitempty"`
```

Comment:
I didn't review the original PR in Evergreen, but is it okay to introduce a dependency on GitHub for this command? It seems like the command could break if GitHub changes its API at all (and would require Shrub users to upgrade to fix it). Usually we rely on built-in types to mitigate that kind of dependency.

If it's intentionally like this, then it seems fine to me.

Reply:
In the original PR in Evergreen, we introduce the direct dependency so that every time GitHub updates, we do not have to update (only the package).

As GitHub adds new permissions, we just have to update the package. If they remove existing permissions for some reason, it would break regardless of the dependency since it gets encoded in to JSON and sent to GitHub all the same (as a map or their struct).

I think adding dependabot is a good idea, I will create a ticket for adding it to shrub and add a section in the ops guide and say it on slack as well if we merge with the dependency.

It's also I think not a priority to upgrade the dependency too, since it won't introduce any security vulnerabilities and at worst-not support new options (which we could just go off of user reports when they use shrub and the option they want is not present) so if we don't get to the update, that's already (unless later we add other dependencies where we use more just a struct for parsing)